### ISSUE TITLE
[question] having trouble loading images from assets

### ISSUE THREAD

**Issue by neelu0** [noties/Markwon#81]
I am having lots of trouble displaying images from assets. at first i was having trouble even loading the drawable loader and would get run time exceptions but i got around it. i know there was a similar issue already addressed it did help me got this far but i am stuck again

```
public class MainActivity extends AppCompatActivity {
    TextView im;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        im=findViewById(R.id.tv1);
        final SpannableConfiguration conf = SpannableConfiguration.builder(getBaseContext())
                .asyncDrawableLoader(AsyncDrawableLoader.create())
                .build();
        Markwon.setMarkdown(im, conf, "![img](assets/twinspri.jpg)");


    }
}

```
with this only the image label gets displayed, i wonder if it is because of the scheme handling or it not being able to find asset folder or its probably a much bigger problem.
i tried using 

```
AsyncDrawableLoader.builder()
 .addSchemeHandler(FileSchemeHandler.createWithAssets(getAssets())
 .resources(getResources)
 .build()
```

 in place of AsyncDrawableLoader.create() but that was crashing the app.
Help would be much appreciated

**Comment by noties** [comment]
Hey @neelu0 !

You actually have multiple options here:
* use an URL like `file:///android_asset/your-file.PNG` in your markdown (note **3** slashes)
* use `UrlProcessorAndroidAssets` which will process all URLs in your markdown and **if** an URL would not have a scheme (http, file, etc) then it will _assume_ that requested resource is from assets

```java
final SpannableConfiguration configuration = SpannableConfiguration.builder(context)
        .urlProcessor(new UrlProcessorAndroidAssets())
```

Add one of these to your existing configuration and check if that works 🙌

**Comment by neelu0** [comment]
hello, thanks for really quick response, i did tried the above solution but it was still showing the alternative text in place of of the image and i even tried the things that were mentioned in previous similar issue already stated https://github.com/noties/Markwon/issues/43 except the .client one
but one thing i tried was adding `.addSchemeHandler(FileSchemeHandler.createWithAssets(getAssets()))` in AsyncDrawableLoader.builder which resulted not displaying anything at all also no runtime exceptions not even alternative text.
```
public class MainActivity extends AppCompatActivity {
    TextView im;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        im=findViewById(R.id.tv1);
        AsyncDrawableLoader ad = AsyncDrawableLoader.builder()
                .resources(Resources.getSystem())
                .addSchemeHandler(FileSchemeHandler.createWithAssets(getAssets()))
                .executorService(Executors.newCachedThreadPool())
                .build();

        final SpannableConfiguration conf = SpannableConfiguration.builder(getBaseContext())
                .asyncDrawableLoader(ad)
                .urlProcessor(new UrlProcessorAndroidAssets())
                .build();
        Markwon.setMarkdown(im, conf, "![twinspri](file:///android_asset/twinspri.jpg)");
```
i also tried without 
 ```
AsyncDrawableLoader ad = AsyncDrawableLoader.builder()
                .resources(Resources.getSystem())
                .addSchemeHandler(FileSchemeHandler.createWithAssets(getAssets()))
                .executorService(Executors.newCachedThreadPool())
                .build();
```
and instead of ad i "did" simple `AsyncDrawableLoader.create()`
but no luck

**Comment by noties** [comment]
Hello @neelu0 !

I have investigated and have found out that there is a bug in the library. It happens when an image is loaded faster than spans are drawn (have known dimensions - canvas width and text size). So, actually images are loaded, but have `[0,0][0,0]` as bounds and thus not showing. I'm marking this as a bug. Fix should be released with the next 2.0.1 release 🙌

**Comment by neelu0** [comment]
oh than it was a bug.
good luck than.
have a nice day too
